pkg/trace/writer/trace.go

```diff
@@ -148,65 +156,49 @@ func (w *TraceWriter) Stop() {
 	w.sender.Stop()
 }
 
-func (w *TraceWriter) handleSampledTrace(sampledTrace *TracePackage) {
-	if sampledTrace == nil || sampledTrace.Empty() {
+func (w *TraceWriter) handleSampledTrace(pkg *TracePackage) {
+	if pkg == nil || pkg.Empty() {
 		log.Debug("Ignoring empty sampled trace")
 		return
 	}
 
-	trace := sampledTrace.Trace
-	events := sampledTrace.Events
+	apiTrace, size := traceutil.APITrace(pkg.Trace)
+	size += pb.Trace(pkg.Events).Msgsize()
```

Comment:
It would be, and initially the code wasn't in `traceutil.APITrace`. I moved it there to reduce the algorithm complexity and the number of iterations through each trace. It increased the speed in the benchmarks by 5%.

Reply:
👍

Reply:
Although honestly I'd prefer your suggestion by far. I think that potentially this 5% is meaningless anyway. 🤷‍♂️

Reply:
> I think that potentially this 5% is meaningless anyway. 🤷‍♂️

Yeah, I tend to lean towards simplicity unless there's a clear reason against it. At `1383 ns/op`, we can run this `1e9/1383 = 723k times per second` before we start hitting a bottleneck :)

Reply:
> Yeah, I tend to lean towards simplicity

I'm always down for that. I've made the change. PTAL

Reply:
```
benchmark                         old ns/op     new ns/op     delta
BenchmarkHandleSampledTrace-4     1117          1373          +22.92%
```